.github/workflows/deploy-to-eks.yaml

```
@@ -0,0 +1,130 @@
+  JUJU_CONTROLLER: kubeflow-controller
+  JUJU_MODEL: kubeflow
+  CKF_CHANNEL: "1.10/stable"
+  MLFLOW_CHANNEL: "latest/stable"
+
+jobs:
+  deploy-solution-to-eks:
+    name: Deploy CKF + MLFlow solution to EKS
+    runs-on: ubuntu-24.04
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set envvars from dependencies.yaml
+        run: |
+          yq eval 'to_entries | .[] | "\(.key)=\(.value)"' ".github/dependencies.yaml" | while IFS= read -r line; do
+            echo "$line" >> "$GITHUB_ENV"
+          done
+
+      - name: Update ENV variables from inputs if available
+        run: |
+          K8S_VERSION=${{ inputs.k8s_version || env.K8S_VERSION }}
+          echo "K8S_VERSION=${K8S_VERSION}" >> $GITHUB_ENV
+          UATS_BRANCH=${{ inputs.uats_branch || env.UATS_BRANCH }}
+          echo "UATS_BRANCH=${UATS_BRANCH}" >> $GITHUB_ENV
+
+      - name: Install CLI tools & dependencies
+        run: |
+          pip install tox
+          sudo snap install juju --channel=${{ env.JUJU_VERSION }}/stable
+          sudo snap install charmcraft --classic
+          sudo snap install terraform  --channel latest/stable --classic
+          juju version
+          terraform --version
+          curl --silent --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_$(uname)_amd64.tar.gz" | sudo tar xz -C /usr/local/bin
+          curl -LO "https://dl.k8s.io/release/v${K8S_VERSION}/bin/linux/amd64/kubectl"
+          chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: ${{ env.REGION }}
+          role-to-assume: ${{ secrets.AWS_ROLE_DEPLOY }}
+          role-session-name: gha-eks-deploy
+
+      - name: Create EKS Cluster
+        run: |
+          eksctl create cluster \
+            --name $CLUSTER_NAME \
+            --region $REGION \
+            --version $K8S_VERSION \
+            --nodegroup-name standard-workers \
+            --node-type t3.large \
+            --nodes 3 \
+            --nodes-min 3 \
+            --nodes-max 5 \
+            --managed
+
+      - name: Update kubeconfig
+        run: aws eks update-kubeconfig --name $CLUSTER_NAME --region $REGION
+
+      - name: Add EKS cluster to Juju
+        run: |
+          juju version
+          juju add-k8s eks --client
+
+      - name: Deploy and assert kubeflow-mlflow solution
+        run: |
+          tox -c ./modules/kubeflow-mlflow -vve test_deployment -- -vv -s
+
+      - name: Run UATs
+        run: |
+          git clone https://github.com/canonical/charmed-kubeflow-uats.git ~/charmed-kubeflow-uats
+          cd ~/charmed-kubeflow-uats
+          git checkout ${{ env.UATS_BRANCH }}
+          tox -e uats-remote -- --filter "not feast"
+
+      # On failure, capture debugging resources
+      - name: Select model
+        run: juju switch aks-controller:kubeflow
```

nitpick: Change the name of the controller (no specific reason for that, just for being clearer), and for all further references to it
```suggestion
        run: juju switch eks-controller:kubeflow
```

---

good catch! obviously a typo from the copy/paste from AKS